src/plugins/testbed/public/plugin.ts

```diff
@@ -17,12 +17,24 @@
  * under the License.
  */
 
-import { Plugin, CoreSetup } from 'kibana/public';
+import { take } from 'rxjs/operators';
+import { Plugin, CoreSetup, PluginInitializerContext } from 'kibana/public';
+
+interface ConfigType {
```

Comment:
👍 for simplicity. 
**optional**: We can add a test that server side types can be reused.
`server/config.ts`
```ts
import { PluginConfigDescriptor } from 'kibana/server';

import { schema, TypeOf } from '@kbn/config-schema';

const configSchema = schema.object({
  secret: schema.string({ defaultValue: 'Not really a secret :/' }),
  uiProp: schema.string({ defaultValue: 'Accessible from client' }),
});

export const config: PluginConfigDescriptor<ConfigType> = {
  exposeToBrowser: ['uiProp'],
  schema: configSchema,
};

export type ConfigType = TypeOf<typeof configSchema>;
export type PublicConfigType = Pick<ConfigType, 'uiProp'>;

```
`server/index.ts`
```ts
import { config, ConfigType } from './config';
export { config };
```
`public/plugin.ts`
```ts
import { PublicConfigType } from '../server/config'; // or re-export it from /server/types.ts
```

Reply:
This is great and ensure the `PublicConfigType` is correctly typed. Only question I have is regarding server-to-client imports: from what I understood, we needs to be extra careful on these imports to avoid polluting the client bundles with imports dependencies, so what should be the recommended way to do this? creates a `types.ts` files at the server plugin root folder level?

Reply:
> creates a types.ts files at the server plugin root folder level?

sounds reasonable. `@kbn/config-schema` is used in node env., so we cannot recommend declaring types under `public` or `common`.